### ISSUE TITLE
UMI_contains_N: xx missing from filtered_parts/*.yaml file

### ISSUE THREAD

**Issue by mbassalbioinformatics** [indrops/indrops#21]
Hi

I have couple of libraries where post running the FILTER step, for some reason some of the filtered_parts/*.yaml files have the line "UMI_contains_N: xx" missing.  This consequently causes the IDENTIFY ABUNDANT BARCODES step to fail.  Im guessing the library quality is good, hence no umis are found that trigger the UMI_N counter, hence its left out of the output yaml file?

Any suggestions how to get around this other than manually inserting this line after the first step then continuing?

**Comment by davidbmorse** [comment]
Hi @mbassalbioinformatics,
Did you manage to solve this issue? I'm having the same problem on a few of my libraries.

**Comment by mbassalbioinformatics** [comment]
> Hi @mbassalbioinformatics,
> Did you manage to solve this issue? I'm having the same problem on a few of my libraries.

Yeah so basically as the files are being processed there is a *<barcode>_metric.yaml file that gets created for each sample.  What ive noticed is that if the sequencing is good, then the umi's dont contain "N"s and a line is not printed to this metric.yaml file.  So all i did to get around it is to manually insert 
UMI_contains_N: 0
into the file at line 9 and restart the pipeline from step 2 and it works fine from there. (Im pretty sure its step 2... might be 3, but pretty sure its 2...)
![contiansn](https://user-images.githubusercontent.com/42388216/52281406-10cb3a00-2999-11e9-9e7c-6c0ca74767ae.PNG)

**Comment by davidbmorse** [comment]
This worked great!
Thank you so much for the help.
Yes, step 2 seems to be the place to restart from after inserting the line in the _metric.yaml file.
